uaclient/entitlements/repo.py

```diff
@@ -106,7 +101,7 @@ def application_status(self) -> "Tuple[ApplicationStatus, str]":
         )
         repo_url = directives.get("aptURL")
         if not repo_url:
-            repo_url = self.repo_url
+            raise exceptions.MissingAptURLDirective(self.name)
```

Comment:
At some point, not today, it might be nice for uaclient to define a set of mandatory keys required of a contract server response. If any are missing, we balk early and in one place instead of handling this in multiple places in the codebase. Now is probably not the time for us to address that. What do you think about a queuing a techdebt issue to tighten up our response validation. Is it worth it?

Reply:
Yeah, I had the same thought. I think a class which represents the response which handles validation in its `__init__` could be a good way to go.

I'm +1 on a techdebt issue.